train_gpt2.cu

```diff
@@ -686,45 +690,61 @@ void gpt2_forward(GPT2 *model, const int* inputs, const int* targets, size_t B,
     }
 
     matmul_forward_cublaslt(acts.output, acts.lnf, params.wte, NULL, B, T, C, Vp, main_stream);
+    cudaCheck(cudaDeviceSynchronize());
+}
 
-    // also forward the cross-entropy loss function if we have the targets
-    if (targets != NULL) {
+
+float gpt2_validate(GPT2 *model) {
```

Comment:
maybe a comment on what gpt2_validate does as a function? what is it validating exactly?

Reply:
yes, the name isn't great. Maybe we should actually move the gpt2_forward call for validation inside here, so that the API allows for a future optimization in which the forward pass doesn't store all activations, and thus can run much larger batch size in validation.